src/tribler-core/tribler_core/session.py

```diff
@@ -249,11 +256,20 @@ def unhandled_error_observer(self, loop, context):
                 text_long = text_long + "\n--LONG TEXT--\n" + buffer.getvalue()
         text_long = text_long + "\n--CONTEXT--\n" + str(context)
 
-        self._logger.error("Unhandled exception occurred! %s", text_long)
+        description = 'session.unhandled_error_observer()'
+        with AllowSentryReports(value=False, description=description):
+            self._logger.error("Unhandled exception occurred! %s", text_long)
+
+        sentry_event = SentryReporter.last_event
+
+        if not self.api_manager:
+            return
+
+        events = self.api_manager.get_endpoint('events')
+        events.on_tribler_exception(text_long, sentry_event)
```

Comment:
@ichorid is it the proper way to transfer exceptions from `core` to `GUI`?

Reply:
Actually, there are two mechanisms for transferring exceptions to the GUI:
* through the `state` endpoint
* through the `events` endpoint

The `events` thing is asynchronous and is the primary way of indicating errors. The `state` thing is synchronous (it is called by `check_core_ready` in the GUI) and it is a redundant way of passing the error to the GUI even in case the `events` endpoint has not connected yet.

So, you should keep both, at least until we come up with a better system.

Reply:
Ok, thank you! Will implement it.

Reply:
Implemented